apps/wing/src/analytics/collect.ts

```diff
@@ -0,0 +1,93 @@
+import { Command } from "commander";
+import { loadAnalyticsConfig, storeAnalyticEvent } from "./storage";
+import { AnalyticEvent } from "./event";
+import { OSCollector } from "./collectors/os-collector";
+import { NodeCollector } from "./collectors/node-collector";
+import { CLICollector } from "./collectors/cli-collector";
+import { CICollector } from "./collectors/ci-collector";
+import { PACKAGE_VERSION } from "../cli";
+
+
+/**
+ * Collects analytics for a given command, stores it for later export
+ * 
+ * @param cmd The commander command to collect analytics for
+ * @returns string the file path of the stored analytic
+ */
+export async function collectCommandAnalytics(cmd: Command): Promise<string | undefined> {
+  const osCollector = new OSCollector();
+  const nodeCollector = new NodeCollector();
+  const ciCollector = new CICollector();
+  const cliCollector = new CLICollector(cmd);
+
+  let event: AnalyticEvent = {
+    event: `wing cli:${cmd.name()}`,
+    properties: {
+      cli: await cliCollector.collect(),
+      os: await osCollector.collect(),
+      node: await nodeCollector.collect(),
+      ci: await ciCollector.collect(),
```

Comment:
We need to flatten this all things under `properties,` nested maps are not good for our analytics consumers 

Instead of:
```
properties: { 
  cli: { 
    k1: "v1", 
    k2: { 
      k3: "v2"
    }
  }
}
```
Is should be:
```
properties: { 
  cli_k1: "v1",
  cli_k2_k3: "v2
}
```

Reply:
also notice that properties should be snake case (just a naming convention)

Reply:
Love the call out. I did not think about consumers querying nested maps aspect. Will fix this conventions